src/Microsoft.Security.Utilities.Core/IdentifiableScan.cs

```diff
@@ -146,268 +69,68 @@ private void PopulateBackupMasker(bool generateCorrelatingIds)
                                                      this.regexEngine);
     }
 
-    public UInt32 PossibleMatches
+    public IEnumerable<Detection> DetectSecrets(string input)
     {
-        get
+        foreach (var hpDetection in highPerformanceSecretScanner.Scan(input))
```

Comment:
>hpDetection [](http://example.com/codeflow?start=21&length=11)

not a fan of this var use here, had to read more code to see this is a detection. another NIT, maybe just `detection` is a fine name. `hp` means hewlett packard to me, their brand is strong.

Hm, but i do see this name collides with the `detection` local below.